### PR TITLE
gunittest: Exclude broken tests

### PR DESCRIPTION
**Pull request by wenzeslaus**
* Adds configuration file to gunittest which can contain list of files to exclude.
* Excluding all currently broken tests.
* Print line with failed test name in pytest style.
* Include .git in hardcoded excluded dirs.
* Make the default min-success 100% (assuming no fluctuations with exclusions).
* No tests found ends with an error message, not traceback.

### CODE REVIEW

File: .gunittest.cfg [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    temporal/t.rast.series/testsuite/test_series.py
+    vector/v.in.lidar/testsuite/decimation_test.py
+    vector/v.in.lidar/testsuite/mask_test.py
+    vector/v.in.lidar/testsuite/test_v_in_lidar_basic.py
+    vector/v.in.lidar/testsuite/test_v_in_lidar_filter.py
+    vector/v.in.pdal/testsuite/test_v_in_pdal_basic.py
+    vector/v.in.pdal/testsuite/test_v_in_pdal_filter.py
+    vector/v.out.lidar/testsuite/test_v_out_lidar.py
+    vector/v.what/testsuite/test_vwhat_layers.py
+    vector/v.what/testsuite/test_vwhat_ncspm.py
```
Comment (neteler) [inline .gunittest.cfg:39]:
```suggestion
    gui/wxpython/core/testsuite/toolboxes.sh
    lib/init/testsuite/test_grass_tmp_mapset.py
    python/grass/gunittest/testsuite/test_assertions_rast3d.py
    python/grass/gunittest/testsuite/test_assertions_vect.py
    python/grass/gunittest/testsuite/test_gmodules.py
    python/grass/gunittest/testsuite/test_gunitest_doctests.py
    python/grass/gunittest/testsuite/test_module_assertions.py
    python/grass/pygrass/raster/testsuite/test_pygrass_raster_doctests.py
    python/grass/pygrass/rpc/testsuite/test_pygrass_rpc_doctests.py
    python/grass/script/testsuite/test_script_doctests.py
    python/grass/temporal/testsuite/unittests_temporal_raster_algebra_equal_ts.py
    python/grass/temporal/testsuite/unittests_temporal_raster_conditionals_complement_else.py
    raster/r.contour/testsuite/testrc.py
    raster/r.geomorphon/testsuite/test_r_geom.py
    raster/r.in.gdal/testsuite/test_r_in_gdal.py
    raster/r.in.lidar/testsuite/test_base_resolution.sh
    raster/r.in.lidar/testsuite/test_base_resolution.sh
    scripts/g.search.modules/testsuite/test_g_search_modules.py
    temporal/t.connect/testsuite/test_distr_tgis_db_raster3d.py
    temporal/t.connect/testsuite/test_distr_tgis_db_raster.py
    temporal/t.connect/testsuite/test_distr_tgis_db_vector.py
    temporal/t.info/testsuite/test.t.info.sh
    temporal/t.rast.aggregate/testsuite/test_aggregation_relative.py
    temporal/t.rast.series/testsuite/test_series.py
    vector/v.in.lidar/testsuite/decimation_test.py
    vector/v.in.lidar/testsuite/mask_test.py
    vector/v.in.lidar/testsuite/test_v_in_lidar_basic.py
    vector/v.in.lidar/testsuite/test_v_in_lidar_filter.py
    vector/v.in.pdal/testsuite/test_v_in_pdal_basic.py
    vector/v.in.pdal/testsuite/test_v_in_pdal_filter.py
    vector/v.out.lidar/testsuite/test_v_out_lidar.py
    vector/v.what/testsuite/test_vwhat_layers.py
    vector/v.what/testsuite/test_vwhat_ncspm.py
```

(trivial: alpha-sort list)

File: python/grass/docs/src/gunittest_running_tests.rst [hunk 2]
```diff
@@ -78,11 +78,31 @@ use::
...
+Test files which would be otherwise collected and executed can be excluded
+from the testing using the ``exclude`` key in the ``.gunittest.cfg``
+configuration file under the ``.gunittest`` section.
+The value of ``exclude`` is whitespace-separated list of exclude files
+possibly with wildcards (from Python fnmatch).
+Directory separators are converted to ``/`` for the matching.
+Paths are relative to the directory the testing was started from.
+
+For example, to exclude the whole directory ``vector`` and one specific file,
+you would use::
```
Comment (neteler) [inline python/grass/docs/src/gunittest_running_tests.rst:98]:
```suggestion
For example, to exclude the whole directory ``vector`` and one specific
raster file, you would use::
```

**Review by neteler — APPROVED** [review]
Looks good to me.
Like that the PRs also in the grass-addons repo should no longer show failures, right?

File: .gunittest.cfg (hunk 1 above)
Comment (wenzeslaus) [reply .gunittest.cfg]: Done.

File: python/grass/docs/src/gunittest_running_tests.rst (hunk 2 above)
Comment (wenzeslaus) [reply python/grass/docs/src/gunittest_running_tests.rst:98]: Hm, raster file sounds like a GeoTIFF or something, leaving as is.

**Comment by wenzeslaus** [comment]
> Like that the PRs also in the grass-addons repo should no longer show failures, right?

If you mean that after this PR and after creating a similar list for grass-addons repo, the test with master branch won't show(1) any errors, then yes.

To fix the failing tests with the 7.8 branch, we need to either fix a (any) test or lower the required success rate.

(1) The failing tests won't show in the detail of a run. They are there now, but won't be afterwards. You will have to locally enable them or run individual test to see them. The tests with the master branch are currently passing.

**Pull request merged**